Menus/BuyMenuGUI.cpp

```diff
@@ -1208,20 +1208,45 @@ void BuyMenuGUI::Update()
 
         // Get handle to the currently selected item, if any
         GUIListPanel::Item *pItem = m_pShopList->GetItem(m_ListItemIndex);
+        std::string description = "";
 
-        // Show popup info box next to selected item, but only if it has a description
-        string description = "";
-        // Get it from the regular Entitiy preset
-        if (pItem && pItem->m_pEntity && !pItem->m_pEntity->GetDescription().empty())
+        if (pItem && pItem->m_pEntity) {
             description = pItem->m_pEntity->GetDescription();
-        // Show popup info box next to selected module item, but only if it has a description
-        else if (pItem && pItem->m_ExtraIndex >= 0)
-        {
-            const DataModule *pModule = g_PresetMan.GetDataModule(pItem->m_ExtraIndex);
-            if (pModule && !pModule->GetDescription().empty())
+            const Entity* currentItem = pItem->m_pEntity;
```

Comment:
Formatting - for pointer and reference variables we always go with `Type *variable` or `Type &variable`. For pointer and reference methods, it looks like `Type * method()` or `Type & method()`.

Applies throughout your changes.

Reply:
What's the correct case for situations like casting? I'm currently using `dymanic_cast<const Type*>`, for example.

Reply:
You'll wanna do `dynamic_cast<Type *>(thing)` or `dynamic_cast<const Type *>(thing)`